app/models/site.rb

```diff
@@ -203,10 +203,24 @@ def self.clear_anon_cache!
     MessageBus.publish(SITE_JSON_CHANNEL, '')
   end
 
+  def self.welcome_topic_banner_cache_key(user_id)
+    "show_welcome_topic_banner:#{user_id}"
+  end
+
   def self.show_welcome_topic_banner?(guardian)
     return false unless guardian.is_admin?
-    return false unless guardian.user.id == User.first_login_admin_id
+    user_id = guardian.user.id
+
+    show_welcome_topic_banner = Discourse.cache.read(welcome_topic_banner_cache_key(user_id))
+    return show_welcome_topic_banner unless show_welcome_topic_banner.nil?
```

Comment:
```suggestion
    return show_welcome_topic_banner if show_welcome_topic_banner.present?
```

Reply:
We specifically need to check for `nil` here because the value for `show_welcome_topic_banner` can be `false` and `false.present?` will be `false`.